### PR TITLE
Add support to configure `projectFactories` for organization folders

### PR DESCRIPTION
**Pull request by mkobit**
issue: https://issues.jenkins-ci.org/browse/JENKINS-45528

### Notes

- Disabled test because I don't know the right way to implement the clearing use case

### CODE REVIEW

**Comment by daspilker** [comment]
Is this still WIP? I guess by "clearing use case" you mean overriding or clearing the default `WorkflowMultiBranchProjectFactory` project factory. I think it would make sense if calling `projectFactory` would delete any previously set project factories and add the ones being configured with the call. That would make the behavior different to e.g. `steps` which always adds new steps, but e.g. `scm` will also replaces any existing SCM configuration.

File: job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/jobs/OrganizationFolderJob.groovy [hunk 1]
```diff
@@ -16,6 +17,8 @@ class OrganizationFolderJob extends ComputedFolder {
 
     /**
      * Sets the organizations in this folder.
+     *
+     * @since 1.58
```
Comment (daspilker) [inline job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/jobs/OrganizationFolderJob.groovy]: no need to copy the version from class level.

File: job-dsl-core/src/main/resources/javaposse/jobdsl/dsl/jobs/OrganizationFolderJob-template.xml [hunk 2]
```diff
@@ -25,6 +25,8 @@
     <triggers/>
     <navigators/>
     <projectFactories>
-        <org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProjectFactory/>
+        <org.jenkinsci.plugins.workflow.multibranch.WorkflowMultiBranchProjectFactory>
+            <scriptPath>Jenkinsfile</scriptPath>
```
Comment (daspilker) [inline job-dsl-core/src/main/resources/javaposse/jobdsl/dsl/jobs/OrganizationFolderJob-template.xml]: leave the default config as it is for backwards compatibility

**Comment by mkobit** [comment]
Thanks @daspilker , that is the feedback I was looking for. I'll take another stab at this in the next few days.

**Comment by mkobit** [comment]
I think I've made the changes you requested and think this is ready for another round of review.

**Pull request merged**